plugin/pkg/shoot/seedmanager/admission.go

```diff
@@ -143,20 +154,53 @@ func (s *SeedManager) Admit(a admission.Attributes) error {
 }
 
 // determineSeed returns an appropriate Seed cluster (or nil).
-func determineSeed(shoot *garden.Shoot, lister gardenlisters.SeedLister) (*garden.Seed, error) {
-	list, err := lister.List(labels.Everything())
+func determineSeed(shoot *garden.Shoot, seedLister gardenlisters.SeedLister, shootLister gardenlisters.ShootLister) (*garden.Seed, error) {
+	seedList, err := seedLister.List(labels.Everything())
 	if err != nil {
 		return nil, err
 	}
 
-	for _, seed := range list {
-		// We return the first matching seed cluster.
-		if seed.DeletionTimestamp == nil && seed.Spec.Cloud.Profile == shoot.Spec.Cloud.Profile && seed.Spec.Cloud.Region == shoot.Spec.Cloud.Region && seed.Spec.Visible != nil && *seed.Spec.Visible && verifySeedAvailability(seed) {
-			return seed, nil
+	// Determine all candidate seed cluster matching the shoot's cloud and region.
+	candidates := []*garden.Seed{}
```

Comment:
Declare empty slice as
```go
var candidates []*garden.Seed
```
(see https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices)
```suggestion
	var candidates []*garden.Seed
```